evap/staff/tests/test_importers.py

```diff
@@ -385,6 +405,111 @@ def test_replace_consecutive_and_trailing_spaces(self):
         success_messages, __, __ = EnrollmentImporter.process(excel_content, self.semester, None, None, test_run=True)
         self.assertIn("The import run will create 1 courses/evaluations and 3 users", "".join(success_messages))
 
+    def test_existing_course_is_not_recreated(self):
+        existing_course, __ = self.create_existing_course()
+
+        old_course_count = Course.objects.count()
+        old_dict = model_to_dict(existing_course)
+
+        __, warnings, __ = EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertIn(
+            "Course Shake (Schütteln) already exists with identical attributes. Course is not created and users are put into the evaluation of that course.",
+            warnings[ImporterWarning.EXISTS],
+        )
+        expected_course_count = old_course_count + 22
+        self.assertEqual(Course.objects.count(), expected_course_count)
+        existing_course.refresh_from_db()
+        self.assertEqual(old_dict, model_to_dict(existing_course))
+
+    def test_existing_course_degree_is_added(self):
+        existing_course, __ = self.create_existing_course()
+
+        # The existing course exactly matches one course in the import data by default. To create a conflict, the degrees are changed
+        existing_course.degrees.set([Degree.objects.get(name_de="Master")])
+
+        EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertSetEqual(
+            set(existing_course.degrees.all()), set(Degree.objects.filter(name_de__in=["Master", "Bachelor"]))
+        )
+
+    def test_existing_course_users_added_to_evaluation(self):
+        __, existing_course_evaluation = self.create_existing_course()
+
+        self.assertFalse(existing_course_evaluation.participants.exists())
+
+        EnrollmentImporter.process(
+            self.default_excel_content, self.semester, self.vote_start_datetime, self.vote_end_date, test_run=False
+        )
+
+        self.assertIn(
+            UserProfile.objects.get(email="lucilia.manilium@institution.example.com"),
+            existing_course_evaluation.participants.all(),
+        )
```

Comment:
If you want to compact this a bit more: This could be part of `test_existing_course_is_not_recreated` (where we test that the corresponding warning is given) (you might want to change the test name then, maybe `test_existing_course_is_merged` or whatever) (feel free to just resolve if you want to keep it)

Reply:
Yeah I think thats ok. If the merging of the participants breaks it should hopefully be obvious by the `assertIn`, right? (Because one could argue that two things are tested at once and thus its harder to see what actually broke).
One thing I am not sure about: Is it weird that the participants merge is combined with `test_existing_course_is_not_recreated`, while the degree merge has a separate test? I would definitely not combine all three though.